### PR TITLE
Add wikimedia module

### LINKED ISSUE #18: Create a crawler to crawl from Wikimedia
Child issue of #4

### PR DESCRIPTION
**Pull request by dinhanhx**
Read the doc.

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -5,29 +5,24 @@ click==7.1.2
...
 python-slugify==4.0.1
-pytz==2020.4
 requests==2.25.0
 six==1.15.0
 slugify==0.0.1
 soupsieve==2.0.1
 text-unidecode==1.3
 tqdm==4.52.0
-update-checker==0.18.0
 urllib3==1.26.2
```
Comment (Huy-Ngo) [inline backend/requirements.txt:26]: We're using `requests` so what's this for?

File: backend/wikimedia/wikicrawl.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		data = {}
+		data['Image source'] = image_src
+		data['Host'] = "Wikimedia"
+		data['Origin URL'] = url
+		data['Title'] = title
+		data_list.append(data)
+
+	return data_list
+
+
```
Comment (Huy-Ngo) [inline backend/wikimedia/wikicrawl.py]: Please indent with spaces to be consistent with other files.

File: backend/wikimedia/wikicrawl.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	Parameters
+		page : int.
+
+	Returns:
+		A list of dictionaries.
+		Each dictionary has
+			- Image source: an url
+			- Host: "Wikimedia"
+			- Original URL: an url
+			- Title: "Wikimedia <number>"
```
Comment (Huy-Ngo) [inline backend/wikimedia/wikicrawl.py]: image
Can you extract these names instead?

Comment (Huy-Ngo) [reply backend/wikimedia/wikicrawl.py]: Similarly, the original URL should leads to [its own page](https://commons.wikimedia.org/wiki/File:Superficial_basal_cell_carcinoma.jpg)

Comment (Huy-Ngo) [reply backend/wikimedia/wikicrawl.py]: And the image URL should be get from that link, which is of original size instead of thumbnail.

**Review by Huy-Ngo — CHANGES_REQUESTED** [review]
I don't know why but the crawling seems slow. Even 3 images takes about 1-2 minutes.

File: backend/requirements.txt (hunk 1 above)
Comment (dinhanhx) [reply backend/requirements.txt:26]: I don't know too but it comes with kaggle-api, so does urllib3. [setup file](https://github.com/Kaggle/kaggle-api/blob/master/setup.py)

https://github.com/Kaggle/kaggle-api/blob/1733863cc05566dea0abf98378a20a1d052e6a85/setup.py#L34-L41

Comment (Huy-Ngo) [reply backend/requirements.txt:26]: OK then

File: backend/wikimedia/wikicrawl.py [hunk 4]
```diff
@@ -1,45 +1,47 @@
...
+    Parameters
+        result : int.
+
+    Returns:
+        A list of dictionaries.
+        Each dictionary has
+            - Image source: an url
+            - Host: "Wikimedia"
+            - Original URL: an url
+            - Title: "Title"
```
Comment (Huy-Ngo) [inline backend/wikimedia/wikicrawl.py]:
```suggestion
            - Image source: The image address
            - Host: "Wikimedia"
            - Original URL: URL to the web page where the image is found
            - Title: "Title"
```

File: backend/wikimedia/wikicrawl.py [hunk 5]
```diff
@@ -1,45 +1,47 @@
...
+    soup = BeautifulSoup(response.content, "lxml")
+
+    infos = soup.find_all("a","image")
+    data_list = []
+    for info in infos:
+        title = info["href"].replace("/wiki/","")
+        origin_url = "mediawiki.org" + info["href"]
+        data = {}
+        #request to origin url to get full image
+        res = requests.get("http://" + origin_url)
```
Comment (Huy-Ngo) [inline backend/wikimedia/wikicrawl.py]:
```suggestion
        origin_url = "https://commons.wikimedia.org" + info["href"]
        data = {}
        #request to origin url to get full image
        res = requests.get(origin_url)
```

File: backend/wikimedia/wikicrawl.py [hunk 6]
```diff
@@ -1,45 +1,47 @@
...
+        sp = BeautifulSoup(res.content, "lxml")
+        image = sp.find("img")
+        data['Host'] = "Wikimedia"
+        data['Origin URL'] = origin_url
+        data['Title'] = title
+        data['Image source'] = image["src"]
```
Comment (Huy-Ngo) [inline backend/wikimedia/wikicrawl.py:39]: Please check this. For some reason, the image source is lacking `https:` and is just `//upload.wikimedia.org/wikipedia/commons/thumb/1/1a/Superficial_basal_cell_carcinoma.jpg/800px-Superficial_basal_cell_carcinoma.jpg`

**Review by Huy-Ngo — CHANGES_REQUESTED** [review]
Overall it seems good, just lack a few things for the URL. The speed for the crawling is still slow - I've tried `urllib.request.urlopen` instead of `requests.get` and it seems it gets faster, though there's still a noticeable lag.

**Review by Huy-Ngo — APPROVED** [review]
OK LGTM

**Pull request merged**